src/app/attend.tsx

```diff
@@ -0,0 +1,51 @@
+import AttendButton, { AttendButtonState, AttendButtonProps } from "../organisms/attend-button";
+import { useAttendAction, AttendResult } from "../commands/attend-action";
+
+function mapAttendResultToButtonState(attendResult: AttendResult): AttendButtonState {
+  switch (attendResult.type) {
+    case "READY":
+      return "ENABLED";
+    case "AWAITING":
+      return "ENABLED"; // ローディング中でもボタンはENABLEDとして表示
+    case "SUCCESS":
+      return "DONE";
+    case "FAILURE":
+      return "OVERTIME";
+    default:
+      return "ENABLED";
+  }
+}
+
+export default function attendApp() {
+  // useAttendAction フックを使用して状態とアクションを管理
+  const [attendResult, submitAttendAction] = useAttendAction();
+
+  // ボタンの状態を AttendButtonState に基づいて決定
+  const buttonState = mapAttendResultToButtonState(attendResult);
```

Comment:
```suggestion
const BUTTON_STATE_MAP: Record<AttendResult, AttendButtonState> = {
  READY: "ENABLED",
  AWAITING: "ENABLED", // FIXME: AttendButtonState に "DISABLED" を追加すべき
  SUCCESS: "DONE",
  FAILURE: "OVERTIME",
};

export default function attendApp() {
  // useAttendAction フックを使用して状態とアクションを管理
  const [attendResult, submitAttendAction] = useAttendAction();

  const buttonState = BUTTON_STATE_MAP[attendResult];
```
関数よりはマッピングするオブジェクトを用意してたほうが, 簡便かつ網羅的な型検証ができます.
また, 本当は申請中はボタンを無効化しておきたいので, ボタンの状態に「無効化」を追加する FIXME コメントを足しておきましょう. あとでこれについて Issue を作成しておきますね.